lms/djangoapps/instructor/permissions.py

```diff
@@ -42,25 +44,53 @@
 perms[GENERATE_CERTIFICATE_EXCEPTIONS] = is_staff
 perms[GENERATE_BULK_CERTIFICATE_EXCEPTIONS] = is_staff
 perms[GIVE_STUDENT_EXTENSION] = HasAccessRule('staff')
-perms[VIEW_ISSUED_CERTIFICATES] = HasAccessRule('staff') | HasRolesRule('data_researcher')
+# TODO: remove role checks once course_roles is fully implemented and data is migrated
+perms[VIEW_ISSUED_CERTIFICATES] = (
+    HasAccessRule('staff') |
+    HasRolesRule('data_researcher') |
+    HasPermissionRule(CourseRolesPermission.ACCESS_DATA_DOWNLOADS.perm_name)
+)
 # only global staff or those with the data_researcher role can access the data download tab
 # HasAccessRule('staff') also includes course staff
-perms[CAN_RESEARCH] = is_staff | HasRolesRule('data_researcher')
+# TODO: remove role checks once course_roles is fully implemented and data is migrated
+perms[CAN_RESEARCH] = (
+    is_staff |
+    HasRolesRule('data_researcher') |
+    HasPermissionRule(CourseRolesPermission.ACCESS_DATA_DOWNLOADS.perm_name)
+)
 perms[CAN_ENROLL] = HasAccessRule('staff')
 perms[CAN_BETATEST] = HasAccessRule('instructor')
-perms[ENROLLMENT_REPORT] = HasAccessRule('staff') | HasRolesRule('data_researcher')
-perms[VIEW_COUPONS] = HasAccessRule('staff') | HasRolesRule('data_researcher')
+# TODO: remove role checks once course_roles is fully implemented and data is migrated
+perms[ENROLLMENT_REPORT] = (
+    HasAccessRule('staff') |
+    HasRolesRule('data_researcher') |
+    HasPermissionRule(CourseRolesPermission.ACCESS_DATA_DOWNLOADS.perm_name)
+)
+# TODO: remove role checks once course_roles is fully implemented and data is migrated
+perms[VIEW_COUPONS] = (
+    HasAccessRule('staff') |
+    HasRolesRule('data_researcher') |
+    HasPermissionRule(CourseRolesPermission.ACCESS_DATA_DOWNLOADS.perm_name)
+)
 perms[EXAM_RESULTS] = HasAccessRule('staff')
 perms[OVERRIDE_GRADES] = HasAccessRule('staff')
-perms[SHOW_TASKS] = HasAccessRule('staff') | HasRolesRule('data_researcher')
+# TODO: remove role checks once course_roles is fully implemented and data is migrated
+perms[SHOW_TASKS] = (
+    HasAccessRule('staff') |
+    HasRolesRule('data_researcher') |
+    HasPermissionRule(CourseRolesPermission.MANAGE_STUDENTS.perm_name) |
+    HasPermissionRule(CourseRolesPermission.ACCESS_DATA_DOWNLOADS.perm_name) |
+    HasPermissionRule(CourseRolesPermission.ACCESS_INSTRUCTOR_DASHBOARD.perm_name)
```

Comment:
Would `ACCESS_INSTRUCTOR_DASHBOARD` need to have an `and`? No user who is gaining access through CourseRoles should be able to see the instructor dashboard if this value is fase.